docs/articles/apex/refactoring.md

```diff
@@ -15,3 +15,25 @@ Validation of the new symbol name is performed before applying the rename. If th
 If the new name conflicts with an existing identifier name, we try to fully qualify the references to the existing name in contexts where the conflicts exist.
 
 ![GIF showing a renaming conflict](../../images/apex-rename-conflict.gif)
+
+## Refactor: Extract Constant
+
+You can extract valid literal expressions into constants. Literal expressions include: String, Integer, Long, Double, Decimal, and Boolean.
+
+This constant will be instantiated below the class declaration. If the expression is contained within an innerclass, the constant will be instantiated below the innerclass declaration. The new constant will always be defined with modifiers private, static, and final.
+
+At this time, there is no conflict detection for the new constant.
+
+![GIF showing extract constant](../../images/extract-constant.gif)
+
+## Refactor: Extract Local Variable
+
+You can extract valid expression types into local variables. Please note that not all expression types are currently supported.
+
+This new variable will be instantiated above the currently selected line. It will retain the formatting of the expression that is selected.
+
+At this time, there is no conflict detection for the new variable and nested expressions are not supported.
+
+![GIF showing extract local variable basics](../../images/extract-local-variable-basics.gif)
```

Comment:
You could remove this gif. The other one you have for local variables is most interesting.